datafusion/core/tests/sql/expr.rs

```diff
@@ -725,6 +725,164 @@ async fn test_interval_expressions() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_interval_mul_div_float() -> Result<()> {
+    macro_rules! test_mul {
+        ($L:expr, $R:expr, $EXPECTED:expr) => {
+            test_expression!(format!("({}) * ({})", $L, $R), $EXPECTED);
+            test_expression!(format!("({}) * ({})", $R, $L), $EXPECTED);
+        };
+    }
+
+    test_mul!(
+        "0.5",
+        "interval '1 month'",
+        "0 years 0 mons 15 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "1.0",
+        "interval '1 month'",
+        "0 years 1 mons 0 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "1.5",
+        "interval '1 month'",
+        "0 years 1 mons 15 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "2.0",
+        "interval '1 month'",
+        "0 years 2 mons 0 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "-1.5",
+        "interval '1 month'",
+        "0 years -1 mons -15 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_expression!(
+        "1.5 * (interval '1 month') / 1.5",
+        "0 years 0 mons 30 days 0 hours 0 mins 0.000000000 secs"
+    );
+
+    test_mul!(
+        "0.5",
+        "interval '1 day'",
+        "0 years 0 mons 0 days 12 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "1.0",
+        "interval '1 day'",
+        "0 years 0 mons 1 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "1.5",
+        "interval '1 day'",
+        "0 years 0 mons 1 days 12 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "2.0",
+        "interval '1 day'",
+        "0 years 0 mons 2 days 0 hours 0 mins 0.000000000 secs"
+    );
+    test_mul!(
+        "-1.5",
+        "interval '1 day'",
+        "0 years 0 mons -1 days -12 hours 0 mins 0.000000000 secs"
+    );
+    test_expression!(
+        "1.5 * (interval '1 day') / 1.5",
+        "0 years 0 mons 0 days 24 hours 0 mins 0.000000000 secs"
+    );
+
+    test_mul!(
+        "0.5",
+        "interval '1 second'",
+        "0 years 0 mons 0 days 0 hours 0 mins 0.500000000 secs"
+    );
+    test_mul!(
+        "1.0",
+        "interval '1 second'",
+        "0 years 0 mons 0 days 0 hours 0 mins 1.000000000 secs"
+    );
+    test_mul!(
+        "1.5",
+        "interval '1 second'",
+        "0 years 0 mons 0 days 0 hours 0 mins 1.500000000 secs"
+    );
+    test_mul!(
+        "2.0",
+        "interval '1 second'",
+        "0 years 0 mons 0 days 0 hours 0 mins 2.000000000 secs"
+    );
+    // This test prints as -1.-500000000 secs, that looks like a bug in printing
+    // TODO fix it
+    // test_mul!(
+    //     "-1.5",
+    //     "interval '1 second'",
+    //     "0 years 0 mons 0 days 0 hours 0 mins -1.500000000 secs"
```

Comment:
I thought I fixed this.

Reply:
That was fixed in cube-js/cube; maybe it copied the logic from arrow-datafusion and the fix is similar.

https://github.com/cube-js/cube/pull/8358/files

Reply:
This behavior is actually in the arrow-rs dependency, in its make_string_interval_day_time macro.